Traktor S3: Fix hotcue color output (rescue #15054)

Swiftb0y opened the pull request:
Just gave this a quick look and it does look LGTM (I regret not getting to it sooner). Don't have the controller to verify though. Any1 willing to also give this a LGTM?

SamWhited committed:
Traktor S3: Fix hotcue color output

Signed-off-by: Sam Whited <sam@samwhited.com>

ywwg commented:
I will take a look

ywwg commented:
I find the new colors very unpleasant: 
<img width="2478" height="1865" alt="image" src="https://github.com/user-attachments/assets/437fe60f-5143-44d3-965c-4ba14f487fcc" />

So, can we make the default "bright" the "medium" level?  If people want super bright they can change that.

The original PR says "no hotcue is ever shown" but i see them just fine with the existing code, so I'm not sure what they mean...

<img width="2478" height="1865" alt="image" src="https://github.com/user-attachments/assets/61337ecf-9875-4512-968b-4543e53ab206" />

On the other hand, with the new code I no longer have the dim white of an unset hotcue.  I liked that backlight because it allowed me to see the buttons in dark clubs.  Having the lights always off will make them hard to see.  Can we put the default white back for unset cues?

ywwg commented on res/controllers/Traktor Kontrol S3.hid.xml:
```diff
@@ -18,6 +17,28 @@
+      label="Bright LED Level">
+      <value label="Off">0</value>
+      <value label="Low">1</value>
+      <value label="Mid">2</value>
```
let's make mid the default

ywwg replied:
looks like "high" is still default

ywwg commented on res/controllers/Traktor-Kontrol-S3-hid-scripts.js:
```diff
@@ -1522,14 +1521,14 @@ TraktorS3.Deck = class {
-        if (loaded) {
+        const hotCueStatus = engine.getValue(this.activeChannel, `hotcue_${  number  }_status`);
+        let ledValue = this.controller.hid.LEDColors.OFF;
+        // If hotcue is set, change the color.
```
please restore the default dim WHITE for unset cues

ywwg replied:
and I'm used to them being on.  How to resolve the dispute?  Fundamentally there is no "correct" answer, it's a matter of taste.  In the past, we have deferred to the person who owns the controller, which would be me in this case.

ywwg replied:
happy to do so.  thank you!

ywwg replied:
Yup that works great!  And I'm fine if the default hotcue color is Black (OFF)

Swiftb0y replied:
Great. Thanks I'll change it to the mixxx default then.

ywwg reviewed (changes requested):
two small changes

Swiftb0y committed:
feat: Make color of unset hotcue pads configurable (including off)

As requested by @ywwg. Defaults to white since this is apparently
the behavior of Traktor (though Mixxx defaults to off).

ywwg reviewed (approved):
thanks!

Swiftb0y commented:
waiting for another test, squashing the fixups once you give the LGTM.
